DataReport/collectBucketMetricsAndUpdateBucketCapacityInfo.js

```diff
@@ -0,0 +1,132 @@
+const async = require('async');
+
+function isSOSCapacityInfoEnabled(bucketInfo) {
+    /**
+     * a valid SOS CapacityInfo Enabled bucketMD looks like
+     * {
+     *     _capabilities: {
+     *         VeeamSOSApi: {
+     *             SystemInfo: {
+     *                 ...
+     *                 ProtocolCapabilities: {
+     *                     CapacityInfo: true,
+     *                     ...
+     *                 },
+     *             ...
+     *             },
+     *             CapacityInfo: {
+     *                 ...
+     *             },
+     *         }
+     *     },
+     * }
+     */
+    return !!(bucketInfo.getCapabilities() // should exist
+        && bucketInfo.getCapabilities().VeeamSOSApi // should exist
+        && bucketInfo.getCapabilities().VeeamSOSApi.SystemInfo // should exist
+        && bucketInfo.getCapabilities().VeeamSOSApi.SystemInfo.ProtocolCapabilities // should exist
+        && bucketInfo.getCapabilities().VeeamSOSApi.SystemInfo.ProtocolCapabilities.CapacityInfo === true // should be true
+        && bucketInfo.getCapabilities().VeeamSOSApi.CapacityInfo); // should exist
+}
+
+function isValidBucketStorageMetrics(bucketMetric) {
+    return bucketMetric
+    && bucketMetric.usedCapacity
+        && typeof bucketMetric.usedCapacity.current === 'number'
+        && typeof bucketMetric.usedCapacity.nonCurrent === 'number'
+    && bucketMetric.usedCapacity.current > -1
+    && bucketMetric.usedCapacity.nonCurrent > -1;
+}
+
+function isValidCapacityValue(capacity) {
+    return (typeof capacity === 'number' && capacity >= 0);
+}
+
+function collectBucketMetricsAndUpdateBucketCapacityInfo(mongoClient, log, callback) {
+    let bucketMetrics;
+    let dataMetricsLastModified;
+    return async.waterfall([
+        next => mongoClient.readCountItems(log, (err, doc) => {
+            if (err) {
+                log.error('an error occurred during readCountItems', {
+                    error: { message: err.message },
+                });
+                return next(err);
+            }
+            dataMetricsLastModified = doc.lastModified;
+            if (doc && doc.dataMetrics && doc.dataMetrics.bucket) {
+                bucketMetrics = doc.dataMetrics.bucket;
+            }
+            return next();
+        }),
+        next => mongoClient.getBucketInfos(log, (err, bucketInfos) => {
+            if (err) {
+                log.error('error getting bucket list', {
+                    error: err,
+                });
+                return next(err);
+            }
+            return next(null, bucketInfos.bucketInfos);
+        }),
+        (bucketInfoList, next) => async.filterLimit(
+            bucketInfoList,
+            10,
+            (bucketInfo, cb) => cb(null, isSOSCapacityInfoEnabled(bucketInfo)),
+            (err, capacityEnabledBuckets) => {
+                next(err, capacityEnabledBuckets);
+            },
```

Comment:
Are these two steps really needed, while we can check this in the `eachLimit` (while better handling I/Os), and thus ignore if bucket is not SOSApi-enabled?